### PR TITLE
Remove run first column

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Removes `runFirstColumn` from the `@cypher` directive, forcing the directive to be used with the `columnName` option.

Updates all tests to comply with this new requirement

## Complexity

Complexity: Medium

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
| 🟥 connections.ConnectionWithSortAndCypher | 14278 | 2277 | 86 | 293 | 2174 |
| 🟥 cypher-directive.TopLevelCypherDirective | 127994 | 99525 | 472 | 772 | 10168 |
| 🟥 query.QueryWithNestedIn | 16823 | 14470 | 58 | 80 | 1635 |
| 🟥 sorting.SortMultipleTypesWithCypherWithCypher | 13401 | 1438 | 198 | 259 | 2174 |
| 🟥 sorting.SortWithTopLevelCypher | 14120 | 2119 | 49 | 177 | 2174 |


🟥 - Performance worsened (dbHits)
🟩 - Performance improved (dbHits)
🟦 - New test
        

<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 34 | 53 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 94 | 181 | 2174 |
|  aggregations.AggregationWithWhere | 13210 | 13210 | 81 | 86 | 2174 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 57 | 82 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 83 | 122 | 2174 |
|  aggregations.DeeplyNestedCount | 10054408 | 10054408 | 4344 | 4260 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 96 | 121 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 283 | 244 | 3003 |
|  connections.Connection | 13042 | 13042 | 167 | 167 | 2174 |
|  connections.NestedConnection | 33883 | 33883 | 121 | 169 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 56 | 89 | 1040 |
| 🟥 connections.ConnectionWithSortAndCypher | 14278 | 2277 | 86 | 293 | 2174 |
|  create.SimpleMutation | 6 | 6 | 28 | 35 | 1 |
| 🟥 cypher-directive.TopLevelCypherDirective | 127994 | 99525 | 472 | 772 | 10168 |
|  delete.SimpleDelete | 18361 | 18361 | 238 | 275 | 1040 |
|  delete.NestedDeleteInUpdate | 16779 | 16779 | 149 | 173 | 2040 |
|  query.SimpleQuery | 14120 | 14120 | 109 | 84 | 2174 |
|  query.QueryWhere | 12992 | 12981 | 42 | 46 | 2165 |
|  query.SimpleQueryWithNestedWhere | 13163 | 13152 | 61 | 77 | 2165 |
|  query.Nested | 10084290 | 10084290 | 9914 | 10272 | 2008534 |
|  query.NestedWithFilter | 38198 | 38198 | 172 | 158 | 4511 |
|  query.OrFilterOnRelationships | 52079 | 51981 | 181 | 214 | 1887 |
|  query.OrFilterOnRelationshipsAndNested | 39208 | 38325 | 222 | 440 | 1887 |
| 🟥 query.QueryWithNestedIn | 16823 | 14470 | 58 | 80 | 1635 |
|  query.NestedConnectionWhere | 13182 | 13182 | 55 | 66 | 2174 |
|  query.DeeplyNestedConnectionWhere | 13284 | 13337 | 83 | 113 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6913 | 6913 | 128 | 157 | 1134 |
|  query.DeeplyNestedWithRelationshipSingleFilters | 15438 | 15438 | 90 | 98 | 2174 |
|  query.Fulltext | 96 | 96 | 26 | 27 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 41 | 44 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 103 | 67 | 1040 |
| 🟥 sorting.SortMultipleTypesWithCypherWithCypher | 13401 | 1438 | 198 | 259 | 2174 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 116 | 119 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 141 | 182 | 4516 |
| 🟥 sorting.SortWithTopLevelCypher | 14120 | 2119 | 49 | 177 | 2174 |
|  unions.SimpleUnionQuery | 321 | 321 | 52 | 67 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 52 | 94 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 303 | 367 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 351 | 474 | 33033 |

</details>
        
Old Schema Generation: 25.999s
Schema Generation: 25.647s

**Comment by Liam-Doodson** [comment]
I've not had a chance to properly look over this yet but a general observation is that I think this could do with some updates to the documentation for the `@cypher` directive. Currently, most of the examples there do not use `columnName`.

The 4.0.0 migration guide also probably needs an update to make clear that at the point this is released, `columnName` will be required

**Comment by angrykoala** [comment]
Yeah, these will be added as a separate PR to reduce the size of these for review

**Review by Liam-Doodson — COMMENTED** [review]
This is looking good, especially the performance improvement!. Just a couple of typos in the updated tests and some edge cases that I'm not clear on the expected behaviour (mainly because I didn't see the first stage of this) and could possibly benefit from tests

**Review by darrellwarde — APPROVED** [review]
The changes are good! You are going to have to remind me, why is it that we decided against running `EXPLAIN` on the queries? I'm still really not 100% a fan of this approach, it feels somewhat like we're pinning our shortcomings onto our users.

File: packages/graphql/tests/integration/directives/cypher/cypher.int.test.ts [hunk 1]
```diff
@@ -79,7 +79,8 @@ describe("cypher", () => {
                                 statement: """
                                 MATCH (m:${Movie} {title: $title})
                                 RETURN m
-                                """
+                                """,
+                                columnName: "m"
```
[the remark this answers is not in the document]
Comment (darrellwarde) [reply packages/graphql/tests/integration/directives/cypher/cypher.int.test.ts:83]: `columnName` is now be a mandatory field in the directive definition, so it will literally fail type definition validation if it isn't passed in.

File: packages/graphql/src/schema/validation/validate-document.test.ts [hunk 2]
```diff
@@ -435,6 +445,7 @@ describe("validateDocument", () => {
                             ORDER BY jaccard DESC
                             RETURN b LIMIT 1
                             """
+                            columnName: "jaccard"
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/src/schema/validation/validate-document.test.ts:448]: good catch!

File: packages/graphql/tests/integration/directives/cypher/cypher.int.test.ts [hunk 3]
```diff
@@ -79,7 +79,8 @@ describe("cypher", () => {
...
                             )
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/tests/integration/directives/cypher/cypher.int.test.ts:84]: it would get ignored, as columnName is used in with

File: packages/graphql/tests/tck/issues/1139.test.ts [hunk 4]
```diff
@@ -1,115 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/tests/tck/issues/1139.test.ts:1]: This test is no longer relevant, as this is only caused when a cypher query does not return anything, which is no longer supported

**Comment by angrykoala** [comment]
answering to you @darrellwarde 

`EXPLAIN` has a lots of gotchas and pitfalls in its implementation, but most importantly, not all of the RETURN cases are covered, as explain does not really produce a consistent list of returned columns, with several cases requiring flaky custom parsing, and other cases simply not providing the variables.

I have the same concerns, and I feel that we could even keep providing a version without columnName using APOC, even if not recommended.

Another alternative could be trying to do some parsing if columnName is not provided. Even if this alternative does not work for all cases, columnName will always be available there to custom configure.

I'm happy to keep this PR open a bit longer if we want to keep this discussion going before merging
